[nnc] Do not try to vectorize kernels that use float16

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55970 [nnc] Do not try to vectorize kernels that use float16**

Summary: LLVM's support for float16 is not great, and we were seeing assertion
failures trying to generate code for vectorized uses.  I note that clang
doesn't even try to vectorize operations involving half:
https://gcc.godbolt.org/z/86MW4xr17, so that's a good sign we shouldn't either.

Fixes #55905

Test Plan: pytest test_jit_fuser_te.py -k test_isnan

Differential Revision: [D27752279](https://our.internmc.facebook.com/intern/diff/D27752279)

bertmaher committed:
[nnc] Do not try to vectorize kernels that use float16

Summary: LLVM's support for float16 is not great, and we were seeing assertion
failures trying to generate code for vectorized uses.  I note that clang
doesn't even try to vectorize operations involving half:
https://gcc.godbolt.org/z/86MW4xr17, so that's a good sign we shouldn't either.

Fixes #55905

Test Plan: pytest test_jit_fuser_te.py -k test_isnan

[ghstack-poisoned]

bertmaher committed:
Update on "[nnc] Do not try to vectorize kernels that use float16"


Summary: LLVM's support for float16 is not great, and we were seeing assertion
failures trying to generate code for vectorized uses.  I note that clang
doesn't even try to vectorize operations involving half:
https://gcc.godbolt.org/z/86MW4xr17, so that's a good sign we shouldn't either.

Fixes #55905

Test Plan: pytest test_jit_fuser_te.py -k test_isnan

Differential Revision: [D27752279](https://our.internmc.facebook.com/intern/diff/D27752279)

[ghstack-poisoned]

malfet commented:
Please add git revert https://github.com/pytorch/pytorch/commit/c47cc30bf5236bcbe30d5b31b15215c1783a8528  as part of this PR

bertmaher committed:
Re-enable fp16 tests on "[nnc] Do not try to vectorize kernels that use float16"


Summary: LLVM's support for float16 is not great, and we were seeing assertion
failures trying to generate code for vectorized uses.  I note that clang
doesn't even try to vectorize operations involving half:
https://gcc.godbolt.org/z/86MW4xr17, so that's a good sign we shouldn't either.

Fixes #55905

Test Plan: pytest test_jit_fuser_te.py -k test_isnan

Differential Revision: [D27752279](https://our.internmc.facebook.com/intern/diff/D27752279)

[ghstack-poisoned]

bertmaher commented:
> Please add git revert [c47cc30](https://github.com/pytorch/pytorch/commit/c47cc30bf5236bcbe30d5b31b15215c1783a8528) as part of this PR

Done
